src/modules/musig/main_impl.h

```diff
@@ -561,14 +602,24 @@ int secp256k1_musig_partial_sig_verify(const secp256k1_context* ctx, const secp2
     /* Multiplying the messagehash by the musig coefficient is equivalent
      * to multiplying the signer's public key by the coefficient, except
      * much easier to do. */
-    secp256k1_musig_coefficient(&mu, session->pk_hash, signer->index);
+    secp256k1_musig_coefficient(&mu, session->keygen_aux.pk_hash, signer->index);
     secp256k1_scalar_mul(&e, &e, &mu);
 
     if (!secp256k1_pubkey_load(ctx, &rp, &signer->nonce)) {
         return 0;
     }
+    /* Verify against negated public key if the combined key was negated. */
+    pubkey_tmp = *pubkey;
+    if (session->keygen_aux.pubkey_is_negated) {
+        secp256k1_ge p;
+        if (!secp256k1_pubkey_load(ctx, &p, &pubkey_tmp)) {
```

Comment:
On a second thought... Have you considered adding a `nonce_is_negated` argument to `schnorrsig_real_verify` ? This would save the load/save cycle here.

Reply:
Thanks for bringing this up. It's actually even easier. We can just negate `e`. Pushed.